crates/uv/tests/it/python_pin.rs

```diff
@@ -198,6 +199,131 @@ fn python_pin() {
     }
 }
 
+#[test]
+fn python_pin_use_global_if_no_local() -> Result<()> {
+    let context: TestContext = TestContext::new_with_versions(&["3.11", "3.12"]);
+
+    let xdg = assert_fs::TempDir::new().expect("Failed to create temp dir");
+    let uv = xdg.child("uv");
+
+    // Without arguments, we attempt to read the current pin (which does not exist yet)
+    uv_snapshot!(context.filters(), context.python_pin().env(EnvVars::XDG_CONFIG_HOME, xdg.path()), @r###"
+    success: false
+    exit_code: 2
+    ----- stdout -----
+
+    ----- stderr -----
+    error: No pinned Python version found
+    "###);
+
+    let versions = uv.child(".python-version");
+    versions.write_str("3.11")?;
+
+    // If no local pin, use global.
+    uv_snapshot!(context.filters(), context.python_pin().env(EnvVars::XDG_CONFIG_HOME, xdg.path()), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+    3.11
+
+    ----- stderr -----
+    "###);
```

Comment:
Should this just be a test case in the subsequent test? The above "No pinned Python version found" case is duplicated and this case seems testable there?

Reply:
Cleaned these tests up to narrow what each one does.